python/dolfinx/wrappers/mesh.cpp

```diff
@@ -31,6 +31,48 @@ namespace py = pybind11;
 namespace dolfinx_wrappers
 {
 
+template <typename T>
+void declare_meshtags(py::module& m, std::string type)
+{
+  std::string pyclass_name = std::string("MeshTags_") + type;
+  py::class_<dolfinx::mesh::MeshTags<T>,
+             std::shared_ptr<dolfinx::mesh::MeshTags<T>>>(
+      m, pyclass_name.c_str(), "MeshTags object")
+      .def(py::init([](const std::shared_ptr<const dolfinx::mesh::Mesh>& mesh,
+                       int dim, const py::array_t<std::int32_t>& indices,
+                       const py::array_t<T>& values) {
+        std::vector<std::int32_t> indices_vec(indices.data(),
+                                              indices.data() + indices.size());
+        std::vector<T> values_vec(values.data(), values.data() + values.size());
+        return std::make_unique<dolfinx::mesh::MeshTags<T>>(
+            mesh, dim, std::move(indices_vec), std::move(values_vec));
+      }))
+      .def_readwrite("name", &dolfinx::mesh::MeshTags<T>::name)
+      .def_property_readonly("dim", &dolfinx::mesh::MeshTags<T>::dim)
+      .def_property_readonly("mesh", &dolfinx::mesh::MeshTags<T>::mesh)
+      .def("ufl_id", &dolfinx::mesh::MeshTags<T>::id)
+      .def_property_readonly("values",
+                             [](dolfinx::mesh::MeshTags<T>& self) {
+                               return py::array_t<T>(self.values().size(),
+                                                     self.values().data(),
+                                                     py::none());
+                             })
```

Comment:
This need an extra argument to tie the lifetime of the `MeshTags` object to the return array.

Reply:
OK, I just copied the existing code. Will add reference_internal.